src/stores/sync/wpcom-sites.ts

```diff
@@ -0,0 +1,192 @@
+	};
+}
+
+function transformSitesResponse( sites: unknown[], connectedSiteIds: number[] ): SyncSite[] {
+	const validatedSites = sites.reduce< SitesEndpointSite[] >( ( acc, rawSite ) => {
+		try {
+			const site = sitesEndpointSiteSchema.parse( rawSite );
+			return [ ...acc, site ];
+		} catch ( error ) {
+			Sentry.captureException( error );
+			return acc;
+		}
+	}, [] );
+
+	const allStagingSiteIds = validatedSites.flatMap( ( site ) => {
+		return site.options?.wpcom_staging_blog_ids ?? [];
+	} );
+
+	return validatedSites
+		.filter( ( site ) => ! site.is_a8c )
+		.filter( ( site ) => ! site.is_deleted || connectedSiteIds.some( ( id ) => id === site.ID ) )
+		.map( ( site ) => {
+			// The API returns the wrong value for the `is_wpcom_staging_site` prop while staging sites
+			// are being created. Hence the check in other sites' `wpcom_staging_blog_ids` arrays.
+			const isStaging = allStagingSiteIds.includes( site.ID );
+			const syncSupport = getSyncSupport( site, connectedSiteIds );
+
+			return transformSingleSiteResponse( site, syncSupport, isStaging );
+		} );
+}
+
+export const wpcomSitesApi = createApi( {
+	reducerPath: 'wpcomSitesApi',
+	baseQuery: fetchBaseQuery(),
+	tagTypes: [ 'WpComSites' ],
+	endpoints: ( builder ) => ( {
+		getWpComSites: builder.query< SyncSite[], { connectedSiteIds: number[]; userId?: number } >( {
+			queryFn: async ( { connectedSiteIds } ) => {
+				const wpcomClient = getWpcomClient();
+				if ( ! wpcomClient ) {
+					return { error: { status: 401, data: 'Not authenticated' } };
+				}
+
+				try {
+					const allConnectedSites = await getIpcApi().getConnectedWpcomSites();
+
+					const fields = [
+						'name',
+						'ID',
+						'URL',
+						'plan',
+						'capabilities',
+						'is_wpcom_atomic',
+						'options',
+						'jetpack',
+						'is_deleted',
+						'is_a8c',
+						'hosting_provider_guess',
+						'environment_type',
+					].join( ',' );
+
+					const response = await wpcomClient.req.get(
+						{
+							apiNamespace: 'rest/v1.2',
+							path: `/me/sites`,
+						},
+						{
+							fields,
+							filter: 'atomic,wpcom',
+							options: 'created_at,wpcom_staging_blog_ids',
+							site_activity: 'active',
+						}
+					);
+
+					const parsedResponse = sitesEndpointResponseSchema.parse( response );
+
+					// First transformation using all connected sites (for reconciliation)
+					const syncSitesForReconciliation = transformSitesResponse(
+						parsedResponse.sites,
+						allConnectedSites.map( ( { id } ) => id )
+					);
+
+					// whenever array of syncSites changes, we need to update connectedSites to keep them updated with wordpress.com
+					const { updatedConnectedSites } = reconcileConnectedSites(
+						allConnectedSites,
+						syncSitesForReconciliation
+					);
+					await getIpcApi().updateConnectedWpcomSites( updatedConnectedSites );
+
+					// Second transformation using connectedSiteIds parameter (for syncSupport calculation for selected site)
```

Comment:
```suggestion
					const syncSitesForReconciliation = transformSitesResponse(
						parsedResponse.sites,
						allConnectedSites.map( ( { id } ) => id )
					);

					const { updatedConnectedSites } = reconcileConnectedSites(
						allConnectedSites,
						syncSitesForReconciliation
					);
					await getIpcApi().updateConnectedWpcomSites( updatedConnectedSites );

```
Nit, but the variable names are clear enough here that I would argue we could skip the line comments.

Reply:
Done as part of [99a6b55](https://github.com/Automattic/studio/pull/2132/commits/99a6b5566847f6ce7851a36c476622bede2d91cf)